packages/web-pkg/src/components/AppTopBar.vue

```diff
@@ -1,24 +1,52 @@
 <template>
-  <div class="oc-flex oc-p-s app-top-bar">
-    <oc-resource
-      v-if="resource"
-      id="app-top-bar-resource"
-      :is-thumbnail-displayed="false"
-      :resource="resource"
-    />
-    <div v-else />
-    <div>
-      <slot name="right" />
-      <oc-button
-        id="app-top-bar-close"
-        :aria-label="$gettext('Close')"
-        size="small"
-        @click="$emit('close')"
-      >
-        <oc-icon name="close" size="small" />
-      </oc-button>
+  <portal to="app.runtime.header.left">
+    <div class="oc-app-top-bar oc-p-xs oc-flex oc-flex-center">
+      <div class="open-file-bar oc-flex oc-mr-s">
+        <oc-resource
+          v-if="resourceReady"
+          id="open-file-bar-resource"
+          :is-thumbnail-displayed="false"
+          :resource="resource"
+        />
+        <oc-resource v-else :resource="{ name: 'Loading' }" />
```

Comment:
What to do with the type error here? The `<oc-resource />` expects a full resource

Reply:
Decided to try and go with a `<span>` and give it the `.oc-resource` class for styling/positioning purposes